File: MongoClientLibrary/Models/ChartWindow.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace MongoClientLibrary.Models;

public class ChartWindow : Window
{
    public ChartWindow(string title, ChartType chartType) : base(title)
    {
        ChartType = chartType;
    }
    
    public ChartType ChartType { get; set; }
}

File: NoSqlDemo/ClientApp/src/app/workspace/models/chartWindow.ts
import {WindowModel} from './window.model';
import {ChartType} from './chartType';

export class ChartWindow implements WindowModel {
  constructor(id: string,
              position: [number, number],
              title: string,
              chartType: ChartType) {
    this.id = id;
    this.position = position;
    this.title = title;
    this.chartType = chartType;
  }

  id: string;
  position: [number, number];
  title: string;

  chartType: ChartType;

}
